core/src/com/fourinachamber/fortyfive/map/detailMap/MapEvent.kt

```diff
@@ -222,3 +231,44 @@ class NPCMapEvent(val npc: String) : MapEvent() {
 
 
 }
+
+/**
+ * event that opens a shop where the player can buy up to 8 cards
+ * @param type which type the restrictions are
+ * @param biome in what biome the shop is
+ */
+class ShopMapEvent(
+    val type: String,
+    private val biome: String,
+    val person: String,
+    val seed: Long,
+    val boughtIndices: MutableList<Int>
+) : MapEvent() {
+
+    override var currentlyBlocks: Boolean = false
+    override var canBeStarted: Boolean = true
+    override var isCompleted: Boolean = false
+
+    override val displayDescription: Boolean = true
+
+    override val descriptionText: String = "Enter shop"
+    override val displayName: String = "BUY STUFF NOW"
+
+    override fun start() {
+        FortyFive.changeToScreen("screens/shop_screen.onj", this) // TODO: ugly
+    }
+
+    fun complete() {
+        canBeStarted = true
+    }
+
+    override fun asOnjObject(): OnjObject = buildOnjObject {
+        name("ShopMapEvent")
+        includeStandardConfig()
```

Comment:
includeStandardConfig is not necessary here because the event can't be completed and can't block